src/Haskell.hs

```diff
@@ -0,0 +1,28 @@
+module Haskell (
+  isModule
+, isQualifiedIdentifier
+, isIdentifier
+) where
+
+import           Data.Char
+
+isModule :: [String] -> Bool
+isModule name = (not . null) name && all isModuleName name
+
+isModuleName :: String -> Bool
+isModuleName name = case name of
+  x : xs -> isUpper x && all isIdChar xs
+  _ -> False
+
+isQualifiedIdentifier :: [String] -> Bool
+isQualifiedIdentifier name = case reverse name of
+  x : xs  -> isIdentifier x && isModule xs
+  _ -> False
+
+isIdentifier :: String -> Bool
```

Comment:
Hmm, but the it would become `isQualifiedVariableIdentifier`.  I think I will ignore the ambiguity in naming for now until we actual have a name clash.